### DISCUSSION TITLE
Reading json array-like but not array

### DISCUSSION THREAD

**Discussion by tawmoto** [nlohmann/json#2433] [Q&A] [answered]
Hello,

I have this json which contains a json-type array, but it does not seem an array.
```
{
  "animals":{
    "dog":{
      "description":"This is a dog",
      "name":"Rex"
    },
    "cat":{
      "description":"This is a cat",
      "name":"Maru"
    }
  }
}
```

I am reading it using this piece of code, after I read the documentation here: https://github.com/nlohmann/json#stl-like-access
```
Json json = Json::parse(str);

Json ss = json["animals"];

for (auto const& node : ss.items())
{
    auto animal = node.key(); // dog or cat

    auto description = node.value()["description"].get<std::string>();
    auto name = node.value()["name"].get<std::string>();
}
```

Is there an alternative using range-loops withount using `ss.items()` and `key/value`?

I am trying to do this
```
for (node& element : ss) 
{
// how can I read the animal type? (dog/cat)
// element.get<std::string>() does not work
}
```

Thank you

**Comment by gregmarr** [comment, marked as answer]
There is no array in that JSON. Arrays are marked with `[]` characters.  You just have objects and strings.

The first loop you posted is the proper way to loop over JSON objects when you need to know the keys.  Why are you trying to remove `.items()` and `key/value`?

**Reply by tawmoto** [reply]
> There is no array in that JSON. Arrays are marked with `[]` characters. You just have objects and strings.
> 
> The first loop you posted is the proper way to loop over JSON objects when you need to know the keys. Why are you trying to remove `.items()` and `key/value`?

Thank you for the answer.
Indeed, is not a array, that's why I written that title :-)
I am not trying to remove the `.items()` and `key/value`, I thought that there is a more elegant solution.

But is ok, I will keep this solution, thanks!
